src/sigreturn.c

```diff
@@ -16,7 +16,7 @@
 #endif
 
 /* The following function might be unused, hence the inline qualifier.  */
-static inline void
+static inline void __attribute__((unused))
```

Comment:
I suppose this change is not needed, the compiler shouldn't issue any warnings for static inline functions.
On the contrary, the compiler will issue a warning if a function marked as unused is actually used.

Reply:
Your supposition is false:
````
 echo 'static inline void foo(){}' | clang -Wall -Werror -c -xc -
<stdin>:1:20: error: unused function 'foo'
      [-Werror,-Wunused-function]
static inline void foo(){}
                   ^
1 error generated.

Reply:
Then this is a compiler bug, a lot of software relies on this static inline thing, please report it to compiler people instead.

Reply:
According to [gcc documentation](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html),
```
-Wunused-function

    Warn whenever a static function is declared but not defined or a non-inline static function is unused. This warning is enabled by -Wall.
```

Sorry, I'm not going to workaround incorrect warnings reported by non-compliant compilers.